mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/VirtualMachine.cs

```diff
@@ -215,6 +214,7 @@ internal void AddThreadToInvalidateList (ThreadMirror threadMirror)
 				//if (threadCache != threads) {//While fetching threads threadCache was invalidated(thread was created/destoyed)
 				//	return GetThreads ();
 				//}
+				threadCache = threads;
```

Comment:
I think you need a call to `Thread.MemoryBarrier()` before this assignment so that all the writes to the threads array are seen by other threads. Otherwise they may still see null. (I guess we only ever run this code on ARM on Mono's CI, but still...)